synom/src/tokens.rs

```diff
@@ -190,6 +190,7 @@ tokens! {
         (pub struct Use                     => "use"),
         (pub struct Where                   => "where"),
         (pub struct While                   => "while"),
+        (pub struct Yield                   => "yield"),
```

Comment:
Should we consider adding more tokens here for other keywords reserved for future use? It seems weird that we had to add this to our tokens list when it was already a reserved keyword in rust.